### PR TITLE
fix: verify min git version on Windows

### LINKED ISSUE #758: fix: installer fails if git is too old.
From a customer:

What's error ```error: unknown option `filter=tree:0'``` when installing ```curl --proto '=https' --tlsv1.2 https://raw.githubusercontent.com/shorebirdtech/install/main/install.sh -sSf | bash```?

we could just remove the filter-tree option if git is too old. Or just remove it always. It’s only an optimization.  We could also just error out if git is too old.  I guess I’d probably just error out.

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Verify that the installed version of git is greater than 2.37.1

Fixes https://github.com/shorebirdtech/shorebird/issues/758

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: install.ps1 [hunk 1]
```diff
@@ -9,6 +9,40 @@ function Test-GitInstalled {
...
+function Compare-SemVer {
+    param (
+        [string]$version1,
+        [string]$version2
+    )
+
+    $version1Components = $version1 -split '\.'
+    $version2Components = $version2 -split '\.'
+
+    for ($i = 0; $i -lt 3; $i++) {
```
Comment (felangel) [inline install.ps1:22]: Does this work with suffixes like 1.0.0-dev.1?

Comment (bryanoltman) [reply install.ps1:22]: Trying this on my windows machine now and it turns out git does not adhere to semver 🙃 The previous implementation worked, so I'm just going to rename the function to reflect what it actually does.

Comment (bryanoltman) [reply install.ps1:22]: (the version installed on my windows laptop is `2.41.0.windows.1`)

File: install.ps1 [hunk 2]
```diff
@@ -23,14 +59,16 @@ function Update-Path {
...
 
 $force = $args -contains "--force"
 
 if (Test-Path $installDirectory) {
     if ($force) {
         Write-Output "Existing Shorebird installation detected. Overwriting..."
         Remove-Item -Recurse -Force $installDirectory
-    } else {
+    }
+    else {
```
Comment (felangel) [inline install.ps1:71]: was this formatting change intentional?

**Review by felangel — APPROVED** [review]

File: install.ps1 (hunk 2 above)
Comment (bryanoltman) [reply install.ps1:71]: Yep. I have my windows machine set up to format powershell on save (not sure why my mac didn't do that), but this is expected and welcome

**Pull request merged**